static/app/utils/profiling/flamegraph.ts

```diff
@@ -351,4 +357,70 @@ export class Flamegraph {
 
     return matches;
   }
+
+  collapseSystemFrames() {
+    const {root, frames, depth} = buildCollapsedFlamegraph(collapse(this.root));
+    this.root = root;
+    this.depth = depth;
+    this.frames = frames;
+  }
+}
+
+const isSystemFrame = (frame: FlamegraphFrame) => !frame.node.frame.is_application;
+
+function collapse(frame: FlamegraphFrame) {
+  frame.children = frame.children.reduce((acc, child) => {
+    acc.push(collapse(child));
+    return acc;
+  }, [] as FlamegraphFrame[]);
```

Comment:
This is just a map right?
```suggestion
  frame.children = frame.children.map(collapse);
```

Reply:
🤦🏼‍♂️